### ISSUE TITLE
[Bug]: [benchmark][cluster][performance] The query performance of two querynodes is similar to that of one querynode

### ISSUE THREAD

**Issue by wangting0128** [milvus-io/milvus#9672]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

**conclusion**：
`Under the condition of Nlist: 2048, index_type: ivf_sq8, nprobe: 8 / 32, cluster, querynode query performance optimization of one node and two nodes is not obvious.`

**perf data**：
![image](https://user-images.githubusercontent.com/26307815/136780678-6b33f678-a332-4ef6-838a-165860652677.png)
![image](https://user-images.githubusercontent.com/26307815/136780666-51db93c9-8298-45c1-b99f-2754d777c0b5.png)

**one querynode**
argo task：benchmark-4tttd
test yaml：
```
client-configmap: client-search-filter-sift50m-ivf-sq8-nlist2048
server-configmap: server-cluster-nolimit
```
server ：
```
NAME: benchmark-4tttd-1
LAST DEPLOYED: Fri Sep 17 08:42:12 2021
NAMESPACE: qa-milvus
STATUS: deployed
REVISION: 1
TEST SUITE: None
NAME                                                   READY   STATUS    RESTARTS   AGE
benchmark-4tttd-1-etcd-0                               1/1     Running   0          86s
benchmark-4tttd-1-milvus-datacoord-bf498d8d5-hkshs     1/1     Running   0          86s
benchmark-4tttd-1-milvus-datanode-556b4fff44-dfhvm     1/1     Running   0          86s
benchmark-4tttd-1-milvus-indexcoord-5c48744d6d-ntnmd   1/1     Running   0          86s
benchmark-4tttd-1-milvus-indexnode-698db79c7c-26j42    1/1     Running   0          86s
uest: GET:https://kubernetes.default.svc.cluster.local/apis/extensions/v1beta1?timeout=32s
benchmark-4tttd-1-milvus-proxy-6ff5df79fb-7r2b9        1/1     Running   0          86s
benchmark-4tttd-1-milvus-pulsar-6ccc5b4dcb-2j9gq       1/1     Running   0          86s
benchmark-4tttd-1-milvus-querycoord-6478f94859-2d9kb   1/1     Running   0          86s
benchmark-4tttd-1-milvus-querynode-55c6f44c5-nsnx9     1/1     Running   0          86s
benchmark-4tttd-1-milvus-rootcoord-58cc6c55d5-75lzp    1/1     Running   0          86s
```

**two querynodes：**
argo task：benchmark-9wrvw
test yaml：
```
client-configmap: client-search-filter-sift50m-ivf-sq8-nlist2048
server-configmap: server-cluster-16c64m-querynode-2
```

server：
```
NAME: benchmark-9wrvw-1
LAST DEPLOYED: Wed Sep 22 04:09:21 2021
NAMESPACE: qa-milvus
STATUS: deployed
REVISION: 1
TEST SUITE: None
NAME                                                   READY   STATUS    RESTARTS   AGE
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /root/.kube/config
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /root/.kube/config
I0922 04:13:51.131776      28 request.go:665] Waited for 1.171371945s due to client-side throttling, not priority and fairness, request: GET:https://kubernetes.default.svc.cluster.l
ocal/apis/authorization.k8s.io/v1beta1?timeout=32s
I0922 04:14:01.331767      28 request.go:665] Waited for 11.371309584s due to client-side throttling, not priority and fairness, request: GET:https://kubernetes.default.svc.cluster.
local/apis/kubeflow.org/v1?timeout=32s
benchmark-9wrvw-1-etcd-0                               1/1     Running   0          4m43s
benchmark-9wrvw-1-milvus-datacoord-6c69699946-r7ffl    1/1     Running   0          4m43s
benchmark-9wrvw-1-milvus-datanode-9f695ccff-lgrpr      1/1     Running   0          4m42s
benchmark-9wrvw-1-milvus-indexcoord-7c7669d68c-5nggb   1/1     Running   0          4m42s
benchmark-9wrvw-1-milvus-indexnode-df466c7d6-77dxk     1/1     Running   0          4m43s
benchmark-9wrvw-1-milvus-proxy-648649d7b5-w5trd        1/1     Running   0          4m43s
benchmark-9wrvw-1-milvus-pulsar-6f5c7ffc8-kxjcv        1/1     Running   0          4m42s
benchmark-9wrvw-1-milvus-querycoord-68cc4889b4-q7g5g   1/1     Running   0          4m43s
benchmark-9wrvw-1-milvus-querynode-7d8d948fbd-97q9v    1/1     Running   0          4m43s
benchmark-9wrvw-1-milvus-querynode-7d8d948fbd-r9fhw    1/1     Running   0          4m43s
benchmark-9wrvw-1-milvus-rootcoord-566678b758-5n4xk    1/1     Running   0          4m43s
```

### Expected Behavior

Under the same conditions, the query performance of two querynodes is twice that of one querynode

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):cluster
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

**Comment by xige-16** [comment]
It may be that the query node processing query message overhead is relatively small in the total query delay. After the performance analysis results come out, we will optimize the query.

**Comment by xiaofan-luan** [comment]
can we verify it on master again?

**Comment by yanliang567** [comment]
@wangting0128 could u please help to reproduce it on master

**Comment by wangting0128** [comment]
**One querynode**
test yaml:
```
client-configmap: client-search-filter-sift50m-ivf-sq8-nlist2048
server-configmap: server-cluster-16c64m
```
test result:

<img width="1838" alt="截屏2021-11-22 18 04 27" src="https://user-images.githubusercontent.com/26307815/142841904-2e6b7113-c603-48c2-9dae-1ce41e8cf5fb.png">

<img width="1838" alt="截屏2021-11-22 18 04 59" src="https://user-images.githubusercontent.com/26307815/142841982-967a6a90-6172-43bf-9cae-cd8b5602b59d.png">

**Two querynodes**
test yaml:
```
client-configmap: client-search-filter-sift50m-ivf-sq8-nlist2048
server-configmap: server-cluster-16c64m-querynode-2
```

test result:

<img width="1833" alt="截屏2021-11-22 18 05 30" src="https://user-images.githubusercontent.com/26307815/142842051-d9094948-1450-450c-80dc-8bb199faea0e.png">


<img width="1835" alt="截屏2021-11-22 18 05 46" src="https://user-images.githubusercontent.com/26307815/142842106-fc439870-9cd3-4801-a3b9-5f535969c913.png">


in conclusion：
When nprobe is 32, search performance is optimized, when nprobe is 8, search optimization is not obvious

@xiaofan-luan @xige-16 @yanliang567

**Comment by xiaofan-luan** [comment]
@wangting0128 very impressive  result.

My guess is that when nprobe is small and result size length, pulsar result written and proxy reduce result dominates the total procedure time cost. since two nodes will return double data to proxy side, proxy takes longer on reduce thus cause the result to be very close to one node.

On the situation of  small data size, small nprobe and large topk, we probably need to add metrics and analysis see what is the bottleneck @czs007

**Comment by wangting0128** [comment]
Following up

**Comment by yanliang567** [comment]
lets keep it
